scripts/completion/dvc.bash

```diff
@@ -10,33 +10,54 @@
+              config index metrics_add metrics_modify metrics_remove move \
+              metrics_show pipeline_list pipeline_show remote_add version \
+              remote_list remote_default'
 
 _dvc_options="-h --help -v --version"
 _dvc_global_options="-h --help -q --quiet -V --verbose"
 
-_dvc_add="-R --recursive -f --file --no-commit"
-_dvc_checkout="$(compgen -G '*.dvc')"
+_dvc_add="-R --recursive -f --file --no-commit $(compgen -G '*')"
+_dvc_cache=""
+_dvc_cache_dir="--global --system --local -u --unset"
+_dvc_checkout="-d --with-deps -R --recursive -f --force $(compgen -G '*.dvc')"
+_dvc_commit="-d --with-deps -R --recursive -f --force $(compgen -G '*.dvc')"
 _dvc_config="-u --unset --local --system --global"
 _dvc_destroy="-f --force"
-_dvc_fetch="--show-checksums -j --jobs -r --remote -a --all-branches -T --all-tags -d --with-deps -R --recursive"
-_dvc_gc="-a --all-branches -T --all-tags -c --cloud -r --remote -f --force -p --project"
-_dvc_import="--resume"
+_dvc_diff="-t --target"
+_dvc_fetch="--show-checksums -j --jobs -r --remote -a --all-branches -T --all-tags -d --with-deps -R --recursive $(compgen -G '*.dvc')"
+_dvc_gc="-a --all-branches -T --all-tags -c --cloud -r --remote -f --force -p --projects -j --jobs"
+_dvc_import="--resume -f --file"
+_dvc_index=""
 _dvc_init="--no-scm -f --force"
 _dvc_install=""
 _dvc_lock="$(compgen -G '*.dvc')"
+_dvc_metrics_add="-t --type -x --xpath $(compgen -G '*')"
 _dvc_metrics=""
+_dvc_metrics_modify="-t --type -x --xpath $(compgen -G '*')"
+_dvc_metrics_remove="$(compgen -G '*')"
+_dvc_metrics_show="-t --type -x --xpath -a --all-branches -T --all-tags -R --recursive $(compgen -G '*')"
 _dvc_move="$(compgen -G '*')"
+_dvc_pipeline_list=""
 _dvc_pipeline=""
-_dvc_pull="--show-checksums -j --jobs -r --remote -a --all-branches -T --all-tags -d --with-deps -f --force -R --recursive"
-_dvc_push="--show-checksums -j --jobs -r --remote -a --all-branches -T --all-tags -d --with-deps -R --recursive"
+_dvc_pipeline_show="-c --commands -o --outs --ascii --dot --tree -l --locked $(compgen -G '*.dvc')" 
+_dvc_pull="--show-checksums -j --jobs -r --remote -a --all-branches -T --all-tags -d --with-deps -f --force -R --recursive $(compgen -G '*.dvc')"
+_dvc_push="--show-checksums -j --jobs -r --remote -a --all-branches -T --all-tags -d --with-deps -R --recursive $(compgen -G '*.dvc')"
+_dvc_remote_add="--global --system --local -d -default -f --force"
+_dvc_remote_default="--global --system --local -u --unset"
+_dvc_remote_list="--global --system --local"
 _dvc_remote=""
-_dvc_remove="--dry -o --outs -p --purge --force"
-_dvc_repro="--dry -f --force -s --single-item -c --cwd -m --metrics -i --interactive -p --pipeline -P --all-pipelines --ignore-build-cache --no-commit -s --single-item"
+_dvc_remote_modify="--global --system --local -u --unset"
+_dvc_remote_remove="--global --system --local"
+_dvc_remove="-o --outs -p --purge -f --force $(compgen -G '*.dvc')"
```

Comment:
If I only have a `something.dvc` file on my working directory, and I do `dvc remove [tab]` does it complete the `something.dvc`, also `dvc remove --o[tab] [tab]` -> `dvc remove --outs something.dvc`?

I'm not sure why I didn't add `compgen` to the rest of the commands, then :sweat_smile:

Reply:
Most of the commands have nothing to do with files so ``compgen`` ain't required there.
Yes, ``dvc remove [tab]`` asks for a [y/n] then shows all the files,bash scripts also.
Yes, your interpretation on ``dvc remove --o [tab] [tab]`` is correct.

Reply:
@Naba7 again, you are talking about how `dvc remove` itself works, but @mroutis is asking about how the completion works.

Reply:
It shows a list of all the files .yaml,.gitignore,.yml,.toml,.md. it shows the list to choose from them.
As, mentioned above ``dvc remove --o[tab] [tab]`` is equivalent to ``dvc remove --outs something.dvc`` as it shows the same list of DVC files and directories. In my system 
```
.appveyor.yml            .gitignore
bin/                     LICENSE
.codeclimate.yml         .pre-commit-config.yaml
.codecov.yml             .pylintrc
CODE_OF_CONDUCT.md       pyproject.toml
.coveragerc              README.rst
.deepsource.toml         scripts/
dvc/                     setup.cfg
.dvc/                    setup.py
.flake8                  tests/
.git/                    .travis.yml
.github/ 
```

Reply:
But `list of DVC files and directories` is not what you've shown in the attachment. It should be any file ending with `.dvc` or `Dvcfile`.

Reply:
My bad. It shows all the files and directories whether be it ``DVC`` or not

Reply:
@Naba7 Hm, maybe using singe quotes will help? Currently we are using double quotes around this, and so, if I understand correctly, it is evaluating it right away, instead of waiting for when it is going to be needed. So maybe something like this would actually suggest .dvc files?
```suggestion
_dvc_remove='-o --outs -p --purge -f --force $(compgen -G \'*.dvc\')'
```

Reply:
@Naba7 Notice that i've edited my comment above. Using that syntax it works as needed. Would you be so kind to update other commands too, while we are at it? :slightly_smiling_face:

Reply:
@efiop Shall I change ``_dvc_options_`` and ``_dvc_global_options`` too?